Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,311 +14,403 @@
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithOrderedSet:(NSOrderedSet*)set range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSOrderedSet alloc] initWithOrderedSet:set range:range copyItems:flag] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithSet:(NSSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSOrderedSet alloc] initWithSet:set] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithSet:(NSSet*)set copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSOrderedSet alloc] initWithSet:set copyItems:flag] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithArray:array range:NSMakeRange(0, [array count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithArray:array range:NSMakeRange(0, [array count]) copyItems:flag];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (array == nil) {
+        return [self init];
+    }
+
+    if (NSMaxRange(range) > [array count]) {
+        [NSException raise:NSInvalidArgumentException
+                    format:@"The given range length is greater than the array length: %u.", [array count]];
+    }
+
+    if (self = [self init]) {
+        NSUInteger start = range.location;
+        NSUInteger end = NSMaxRange(range);
+
+        if (flag) {
+            for (NSUInteger i = start; i < end; i++) {
+                id obj = [[array objectAtIndex:i] copy];
+                [self _insertObject:obj];
+                [obj release];
+            }
+        } else {
+            for (NSUInteger i = start; i < end; i++) {
+                [self _insertObject:[array objectAtIndex:i]];
+            }
+        }
+    }
+    return self;
+}
+
+- (void)_insertObject:(id)object {
+    if ([self containsObject:object]) {
+        return;
+    }
+
+    setContainer.push_back(object);
+    mapContainer[object] = std::prev(setContainer.end());
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithObject:(id)object {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSUInteger)count {
+    return setContainer.size();
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithObjects:(id)firstObj {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSArray*)array {
```

Comment:
You can simplify this some:

return [NSArray arrayWithObjects:setContainer.data() count:setContainer.size()];

And then you don't have to copy the array at the end.